sql/select.go

```diff
@@ -82,3 +82,50 @@ func (p *planner) Select(n *parser.Select) (planNode, error) {
 	}
 	return s, nil
 }
+
+type subqueryVisitor struct {
+	*planner
+	err error
+}
+
+var _ parser.Visitor = &subqueryVisitor{}
+
+func (v *subqueryVisitor) Visit(expr parser.Expr) parser.Expr {
+	if v.err != nil {
+		return expr
+	}
+	subquery, ok := expr.(*parser.Subquery)
+	if !ok {
+		return expr
+	}
+	var plan planNode
+	if plan, v.err = v.makePlan(subquery.Select); v.err != nil {
+		return expr
+	}
+	var rows parser.DTuple
+	for plan.Next() {
+		values := plan.Values()
+		switch len(values) {
+		case 1:
+			// TODO(pmattis): This seems hokey, but if we don't do this then the
+			// subquery expands to a tuple of tuples instead of a tuple of values and
+			// an expression like "k IN (SELECT foo FROM bar)" will fail because
+			// we're comparing a single value against a tuple. Perhaps comparison of
+			// a single value against a tuple should succeed if the tuple is one
+			// element in length.
+			rows = append(rows, values[0])
+		default:
+			valuesCopy := make(parser.DTuple, len(values))
```

Comment:
The result from `planNode.Values()` is only valid until the next call to `planNode.Next()`. I'll add a comment and make this clear in the `planNode` documentation.